remote-trait-object/src/port/client.rs

```diff
@@ -120,12 +122,21 @@ impl Drop for Client {
     }
 }
 
-fn receive_loop(transport_recv: Receiver<Packet>, to_slot_receivers: Vec<Sender<Packet>>) -> Result<(), RecvError> {
+fn receive_loop(transport_recv: Box<dyn TransportRecv>, to_slot_receivers: Vec<Sender<Result<Packet, RecvError>>>) {
     loop {
-        let packet = transport_recv.recv()?;
-        let slot_id = packet.view().slot();
-        to_slot_receivers[slot_id.as_usize()]
-            .send(packet)
-            .expect("Slot receivers are managed in Client. Client must be dropped after this thread");
+        match transport_recv.recv(None) {
+            Ok(x) => {
+                let packet = Packet::new_from_buffer(x);
+                let slot_id = packet.view().slot();
+                to_slot_receivers[slot_id.as_usize()]
+                    .send(Ok(packet))
+                    .expect("Slot receivers are managed in Client. Client must be dropped after this thread");
+            }
+            Err(RecvError::Termination) => return,
+            Err(RecvError::TimeOut) => unreachable!(),
+            Err(_err) => {
+                // TODO: Broadcast the error to all **active** call slots
```

Comment:
Shouldn't we return at this point?
If the recv is closed, this code can make a busy loop that does not do anything meaningful.